RFC-005-SWAP-Basic-Bitcoin.md

```diff
@@ -106,6 +106,62 @@ To compute the exact Bitcoin script bytes of the contract, implementations shoul
 | `refund_identity` | 74                     | 93                    | 20             | See [Parameters](#parameters)                                              |
 | `6888ac`          | 94                     | 96                    | 3              | `OP_ENDIF OP_EQUALVERIFY OP_CHECKSIG`                                      |
 
+
+Script execution overview:
+
+```
+// cur element on stack: byte
+//   evaluate byte: 01
+//   OP_IF (redeem) is entered
+//   pop byte
+OP_IF
+
+    // cur element on stack: secret
+    //   peek secret size and push size on stack
+    //   push constant 32 on stack
+    //   pop 2 element of stack (32, secret size) and check equality
+    OP_SIZE 32 OP_EQUALVERIFY
```

Comment:
Not 100% correct: `OP_EQUALVERIFY` is a combination of `OP_EQUAL` and `OP_VERIFY` meaning under the hood it first executes equal (which consumes the top 2 elements) pushes back the result (0 or 1) and then verifies if the top of the stack != 0.